src/ZenohUTransport.cpp

```diff
@@ -194,7 +194,11 @@ v1::UStatus ZenohUTransport::registerRequestListener_(
 		              zenoh::detail::loan(query));
 
 		query_map_.emplace(std::move(id_str), std::move(cloned_query));
-		listener(queryToUMessage(query));
+		try {
+			listener(queryToUMessage(query));
+		} catch (const std::exception& e) {
+			spdlog::error("query processing failed: {}", e.what());
```

Comment:
What exceptions are possible here? We generally don't want to blanket catch all exceptions as they can mask programmatic mistakes. This should either catch specific exceptions that represent temporary failures (that is, failures where trying again may produce a different result), or catch nothing at all.

Additionally, shouldn't a non-ok status be returned if an exception is caught here? Doesn't that represent a failure that needs to be communicated upward?

Reply:
> What exceptions are possible here? We generally don't want to blanket catch all exceptions as they can mask programmatic mistakes. This should either catch specific exceptions that represent temporary failures (that is, failures where trying again may produce a different result), or catch nothing at all.

deserialization problem etc.

> Additionally, shouldn't a non-ok status be returned if an exception is caught here? Doesn't that represent a failure that needs to be communicated upward?

I had a question about this and there was a suggestion in the previous version of this PR, but the listener as it currently stands doesn't allow it. So, by advice from this comment https://github.com/eclipse-uprotocol/up-transport-zenoh-cpp/pull/88#issuecomment-2266076115, I just log and drop the message.